cg/utils/files.py

```diff
@@ -41,3 +42,43 @@ def rename_file(file_path: Path, renamed_file_path: Path) -> None:
 def is_pattern_in_file_path_name(file_path: Path, pattern: str) -> bool:
     """Check if a pattern is in a file path."""
     return pattern in file_path.name
+
+
+def get_creation_time_stamp(directory_path: Path) -> float:
+    """
+    Return time stamp that a directory is created.
+    Raises:
+        FileNotFoundError if the directory does not exist.
+        ValueError if the specified path is not a directory.
+    """
+    if not directory_path.exists():
+        raise FileNotFoundError(f"Directory with path {directory_path} is not found.")
+
+    if not directory_path.is_dir():
+        raise ValueError(f"Specified path {directory_path} is not a directory.")
+
+    stat_info = os.stat(directory_path)
+    creation_timestamp = stat_info.st_ctime
+    return creation_timestamp
+
+
+def remove_directory_and_contents(directory_path):
+    """
+    Delete a directory and its contents.
+    Raises OSError if not successful.
+    """
+    try:
+        shutil.rmtree(directory_path)
+        LOG.info(f"Successfully removed the directory and its contents: {directory_path}")
+    except OSError as e:
+        LOG.error(f"Failed to remove the directory {directory_path} and its contents: {e}")
```

Comment:
We are excepting the OSError, so the docstring is misleading since we specifically do not raise the OSError. Whether the error should be raised or whether the docstring should be changed is hard for me to say, though.

Reply:
I modified docstring to excepts orerror